pkg/domain/entities/types/images.go

```diff
@@ -147,6 +147,57 @@ type ImagePullReport struct {
 	ID string `json:"id,omitempty"`
 }
 
+// ImagePullReportV2 provides a detailed status of a container image pull operation.
+type ImagePullReportV2 struct {
+	// Status indicates the current state of the image pull operation.
+	// Possible values are:
+	// - "error":   An error occurred during the pull.
+	// - "pulling": The image pull is in progress.
+	// - "success": The image pull completed successfully.
+	Status string `json:"status,omitempty"`
```

Comment:
(Non-blocking: consider defining a separate Go type, to make it impossible to accidentally set `Status` to undocumented values. It would also make it easier to consume the progress data in a `podman-remote` client.)